packages/discord.js/src/util/Util.ts

```diff
@@ -499,10 +505,18 @@ class Util extends null {
    * @returns {Promise<Channel[]|Role[]>} Updated item list, with `id` and `position` properties
    * @private
    */
-  static async setPosition(item, position, relative, sorted, client, route, reason) {
+  static async setPosition(
```

Comment:
Its return type will be mixed of both Channel and Role. No?

imo adding 2 overload types is necessary.